ci_environment/docker/recipes/default.rb

```diff
@@ -11,15 +11,15 @@
 include_recipe 'travis_build_environment'
 
 apt_repository 'docker' do
-  uri 'https://get.docker.io/ubuntu'
-  distribution 'docker'
+  uri 'https://apt.dockerproject.org/repo'
+  distribution 'ubuntu-trusty'
   components ['main']
-  key 'https://get.docker.io/gpg'
+  key 'https://apt.dockerproject.org/gpg'
   action :add
 end
 
-docker_pkg = 'lxc-docker'
-docker_pkg += "-#{node['docker']['version']}" if node['docker']['version']
+docker_pkg = 'docker-engine'
+# TODO docker_pkg += "-#{node['docker']['version']}" if node['docker']['version']
```

Comment:
I'm not sure how to do a version pinning in this framework either, but with this repo it's a standard `reprepro` so we'll only have latest available anyhow.  I'm not sure whether that's acceptable or not.  There's also no history beyond the current 1.7.1, so we couldn't use this to do regression testing across Docker versions without a lot more logic switching this based on `node['docker']['version']`, but I don't know if the nodes are provisioned on the fly either (or this is just used for static image creation).